### PR TITLE
Fix Upsample Decomp Striding For Small Channels

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89528

Fix for https://github.com/pytorch/pytorch/issues/93788. 

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Comment by ngimel** [comment]
But suggest_memory_format for that input shouldn't be channels-last in the first place, because stride at dim 0 is 0? https://github.com/pytorch/torchdynamo/issues/623#issuecomment-1323123782

**Review by ngimel — APPROVED** [review]

**Review by anijain2305 — APPROVED** [review]

**Comment by eellison** [comment]
@ngimel 
```
>>> unsqueeze.stride()
(3, 1, 1920, 3)
>>> unsqueeze.is_contiguous(memory_format=torch.channels_last)
```

**Comment by ngimel** [comment]
Yeah but `is_contiguous` is much more lax than actual `suggest_memory_format`

**Comment by ngimel** [comment]
I can make repro fail with 20 channels, which won't be covered by this fix:
```
from torch.nn import *
class Repro(torch.nn.Module):
    def __init__(self):
        super().__init__()



    def forward(self, arg0_1):
        unsqueeze = torch.ops.aten.unsqueeze.default(arg0_1, 0);  arg0_1 = None
        upsample_bilinear2d = torch.ops.aten.upsample_bilinear2d.default(unsqueeze, [800, 1199], False);  unsqueeze = None
        return (upsample_bilinear2d,)

args = [((20, 427, 640), (1, 640*20, 20), torch.float16, 'cuda')]
args = [rand_strided(sh, st, dt, dev) for (sh, st, dt, dev) in args]
mod = make_fx(Repro())(*args)

from torch._inductor.compile_fx import compile_fx_inner
from torch._dynamo.debug_utils import same_two_models

compiled = compile_fx_inner(mod, args)
compiled(args)
```

**Comment by eellison** [comment]
Filed https://github.com/pytorch/pytorch/issues/89539, and adding skip here to fix this failure
